File: backend/internal/models/debate_stats.go
package models

import "time"

type DebateTopicStats struct {
	Topic            string    `json:"topic"`
	TotalParticipants int      `json:"totalParticipants"`
	TotalAgree       int      `json:"totalAgree"`
	TotalDisagree    int      `json:"totalDisagree"`
	SessionsCount    int      `json:"sessionsCount"`
	LastUpdated      time.Time `json:"lastUpdated"`
}

File: packages/shared/src/domain/debate-stats.ts
/**
 * Debate Stats Domain Models
 *
 * TypeScript interfaces for debate statistics.
 */

export interface DebateTopicStats {
  topic: string;
  totalParticipants: number;
  totalAgree: number;
  totalDisagree: number;
  sessionsCount: number;
  lastUpdated: Date | string;
}

export interface RecordDebateStatsRequest {
  topic: string;
  agreeCount: number;
  disagreeCount: number;
  participants: number;
  debateId?: string; // Optional: track which debate this is for (prevents duplicates)
}
